File: core/core-api/src/main/java/io/say/better/domain/review/ui/ReviewController.java
package io.say.better.domain.review.ui;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import io.say.better.domain.review.application.ReviewFacade;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;

@Tag(name = "Review", description = "Review API")
@RestController
@RequestMapping("/api/review")
@RequiredArgsConstructor
public class ReviewController {

	private final ReviewFacade reviewFacade;

}

File: core/core-api/src/main/kotlin/io/say/better/domain/review/ui/ReviewController.kt
package io.say.better.domain.review.ui

import io.say.better.core.common.response.ResponseDto
import io.say.better.domain.review.application.ReviewFacade
import io.say.better.domain.review.ui.dto.ReviewRequest
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/review")
class ReviewController(
    private val reviewFacade: ReviewFacade,
) {
    @PostMapping
    fun submitReview(
        @RequestBody request: List<ReviewRequest.SubmitReview>,
    ): ResponseDto<Nothing?> {
        reviewFacade.submitReview(request)
        return ResponseDto.onSuccess(null)
    }
}
